### PR TITLE
chore(server): organization of cloud-specific files

### PR DESCRIPTION
**Pull request by emilyrohrbough**
While looking into [23517](https://github.com/cypress-io/cypress/issues/23517) I came across how scattered our code is for communicating with the server.

Also did some small typescript conversions because why-not!

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/user.ts [hunk 1]
```diff
@@ -1,69 +0,0 @@
...
-
-      return cache.removeUser().then(() => {
-        if (authToken) {
-          return api.postLogout(authToken)
-        }
-      })
-    })
-  },
-
-  syncProfile (authToken) {
```
Comment (emilyrohrbough) [inline packages/server/lib/user.ts:32]: This was not being used

File: packages/server/lib/user.ts [hunk 2]
```diff
@@ -1,69 +0,0 @@
...
-        name: res.name,
-        email: res.email,
-      }
-
-      return this.set(user)
-      .return(user)
-    })
-  },
-
-  ensureAuthToken () {
```
Comment (emilyrohrbough) [inline packages/server/lib/user.ts:49]: This was replaced by https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/actions/AuthActions.ts#L35

**Review by BlueWinds — APPROVED** [review]

**Comment by chrisbreiding** [comment]
@emilyrohrbough I think we're moving to calling it Cypress Cloud in the (near?) future, so it's possible "dashboard" will soon be an outdated term. Should we go ahead and use "cloud" instead?

**Comment by emilyrohrbough** [comment]
@chrisbreiding Good idea

**Review by chrisbreiding — APPROVED** [review]
Nice refactor!

File: packages/server/lib/cloud/exception.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-const _ = require('lodash')
+import _ from 'lodash'
 const Promise = require('bluebird')
```
Comment (AtofStryker) [inline packages/server/lib/cloud/exception.ts:2]: should this be cut over to an import as well?

**Review by AtofStryker — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**